modules/dhcp/providers/server/isc.rb

```diff
@@ -288,8 +298,7 @@ def read_config file, ignore_includes=false
       logger.debug "Reading config file #{file}"
       config = []
       File.readlines(file).each do |line|
-
-        line.strip! # remove left and right whitespace
+        line = line.split('#').first.strip # remove comments, left and right whitespace
```

Comment:
once you call `String#split` all instances of the symbol you used to split the string will be removed,  which makes https://github.com/bagasse/smart-proxy/blob/Fixes_12211_-_Improve_DHCP_subnets_parsing_to_get_more_informations_from_DHCP_server/modules/dhcp/providers/server/isc.rb#L302 redundant. 

I'm ok with this change if you remove code on line 302.